fuel-vm/src/tests/crypto.rs

```diff
@@ -578,30 +650,38 @@ fn ed25519_verify_a_gt_vmaxram_sub_64() {
 fn ed25519_verify_b_gt_vmaxram_sub_64() {
     let reg_a = 0x20;
     let reg_b = 0x21;
+    let reg_c = 0x22;
 
     #[rustfmt::skip]
     let script = vec![
         op::xor(reg_b, reg_b, reg_b),
         op::not(reg_a, RegId::ZERO),
         op::subi(reg_a, reg_a, 63),
-        op::ed19(reg_b, reg_a, reg_b),
+        op::movi(reg_c, 32),
+        op::ed19(reg_b, reg_a, reg_b, reg_c),
         op::ret(RegId::ONE),
     ];
 
     check_expected_reason_for_instructions(script, MemoryOverflow);
 }
 
-#[test]
-fn ed25519_verify_c_gt_vmaxram_sub_32() {
+#[test_case(31, 32 => (); "Just over the end with 32 bits")]
+#[test_case(63, 64 => (); "Just over the end with 64 bits")]
+#[test_case(31, 0 => (); "Zero defaults to 32")]
+#[test_case(31, 100 => (); "Way over the end")]
+#[test_case(0, 32 => (); "Empty range, goes over it")]
+fn ed25519_verify_c_plus_d_gt_vmaxram(offset: u16, len: u32) {
```

Comment:
I feel like the name should mention that we should expect it to "overflow". All of these tests have to do with the offset causing the overflow, so maybe the name should reflect that.

The `test_case` names don't really inform me that much either, but I think that has more to do with the base name not being clear enough.

Reply:
The other tests were named like this as well, I was trying to be consistent. But I just fixed them all in eb118178 now.